src/FacilitatorOptionParser.ts

```diff
@@ -0,0 +1,179 @@
+      return new Config(mosaicConfig, facilitatorConfig);
+    }
+
+    if (this.mosaicConfigPath) {
+      const mosaic: MosaicConfig = MosaicConfig.fromFile(this.mosaicConfigPath);
+      this.verifyChainIdInMosaicConfig(mosaic);
+      const facilitator = FacilitatorConfig.from(this.auxChain);
+      return new Config(mosaic, facilitator);
+    }
+
+    const facilitator: FacilitatorConfig = FacilitatorConfig.from(this.auxChain);
+    const mosaic: MosaicConfig = MosaicConfig.fromChain(this.originChain);
+    return new Config(mosaic, facilitator);
+  }
+
+  /**
+   * This method returns config object when facilitator config is provided and
+   * origin chain and aux chain is not provided.
+   * @returns Config object encapsulating facilitator and mosaic configs.
+   */
+  private handleFacilitatorConfigOption(): Config {
+    let configObj = {} as any as Config;
+    // When no origin and aux chain provided.
+    if (this.mosaicConfigPath) {
+      configObj = Config.getConfigFromPath(this.mosaicConfigPath, this.facilitatorConfigPath);
+    } else {
+      const facilitatorConfig: FacilitatorConfig = FacilitatorConfig.fromPath(
+        this.facilitatorConfigPath,
+      );
+      const mosaicConfig: MosaicConfig = MosaicConfig.fromChain(
+        facilitatorConfig.originChainId,
+      );
+      configObj = new Config(mosaicConfig, facilitatorConfig);
+    }
+    return configObj;
+  }
+
+  /**
+   * It verifies chain ids in facilitator config.
+   * @param facilitatorConfig Facilitator object containing facilitator config.
+   */
+  private verifyChainIdInFacilitatorConfig(
+    facilitatorConfig: FacilitatorConfig,
+  ): void {
+    if (facilitatorConfig.chains[this.auxChain] === undefined
+      || facilitatorConfig.chains[this.auxChain] === null
+    ) {
+      throw new FacilitatorStartException(
+        `facilitator config is invalid as provided auxchain ${this.auxChain} is not present`,
+      );
+    }
+
+    if (facilitatorConfig.chains[this.originChain] === undefined
+      || facilitatorConfig.chains[this.originChain] === null
+    ) {
+      throw new FacilitatorStartException(
+        `facilitator config is invalid as provided origin chain ${this.originChain} is not present`,
+      );
+    }
+  }
+
+  /**
+   * It verifies chain ids in mosaic configs.
+   * @param mosaicConfig Mosaic object containing mosaic config.
+   */
+  private verifyChainIdInMosaicConfig(
+    mosaicConfig: MosaicConfig,
+  ): void {
+    if (!mosaicConfig.auxiliaryChains[this.auxChain]) {
+      throw new FacilitatorStartException('aux chain is not present in mosaic config');
+    }
+
+    if (mosaicConfig.originChain.chain !== this.originChain) {
+      throw new FacilitatorStartException('origin chain id in mosaic config is different '
+        + 'than the one provided');
+    }
+  }
+
+  /**
+   * It verifies whether both origin and aux chain ids are defined.
+   */
+  private isOriginAuxChainDefined(): void {
```

Comment:
Function name should be improved. Function names started with `is` are assumed (at least for me subjectively) to return boolean.